### ISSUE TITLE
Security: When opening password protected Excel files, the password is spoken outloud while typing

### ISSUE THREAD

**Issue by XLTechie** [nvaccess/nvda#14529]
### Steps to reproduce:

1. Open/create an Excel worksheet.
2. Password-protect that file (Alt+F, I, P, E; type password, press enter, type password, press enter).
3. Exit Excel.
4. Attempt to open the password-protected file.

### Actual behavior:

Upon opening the file, Excel presents a dialog requesting the password for the file.

As you enter the password, circles appear on the screen, as NVDA will confirm if you use arrow keys to read what you typed.
**NVDA does not treat this as a password entry field, however, and as you type the password, the individual characters are spoken out loud if character echo is turned on.**

### Expected behavior:

NVDA should speak "star" or "circle" during this password entry.

### NVDA logs, crash dumps and other attachments:

* [Log from pressing enter to open the file, through attempted password entry](https://github.com/nvaccess/nvda/files/10388770/nvda-excel-spoken-password-entry.txt)
* [Dev Info for password entry dialog](https://github.com/nvaccess/nvda/files/10388782/nvda-excel-password-entry-devinfo.txt)
* [Full run w/add-ons disabled, and lots more errors to wade through](https://github.com/nvaccess/nvda/files/10389094/nvda-excel-full-run.txt)

#### NVDA:

Installed, alpha-27459,816496f1 (2023.1.0.27459)
 
#### Windows version:

10 Home, 22H2 (OS Build 19045.2364)

#### Name and version of other software in use when reproducing the issue:

Office 2019; 16.0.15831.20208

#### Does the issue still occur after restarting your computer?

Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No, but given that NV Access has known of this since at least 2020 (based on the comment by @Qchristensen in #5221), it has presumably been true for many versions of NVDA and Excel.

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes

**Comment by seanbudd** [comment]
to confirm, is this a well written duplicate of #5221? Can #5221 be closed in favour of this?

**Comment by XLTechie** [comment]
@seanbudd no. Though this was mentioned in a [comment](https://github.com/nvaccess/nvda/issues/5221#issuecomment-628283728) there by @Qchristensen as being a related problem, the base issue is not the same.

Not sure if fixing this one will fix that one, as while there are still reports of it sometimes, #5221 is hard to repro reliably. (I did, once, but only once.)

This one, however, repros every time.

**Comment by Adriani90** [comment]
@XLTechie now that #5221 doesn't occur anymore, can you pleaes test whether you still see this issue with NVDA last 64 bit alpha and Microsoft Office 365 version 2510?
